src/delegation.rs

```diff
@@ -46,7 +52,167 @@ pub struct Delegation<
     pub varsig_header: V,
     pub payload: Payload<DID>,
     pub signature: DID::Signature,
-    _marker: std::marker::PhantomData<C>,
+    _marker: PhantomData<C>,
+}
+
+pub struct DelegationRequired<
+    DID: Did = did::preset::Verifier,
+    V: varsig::Header<C> = varsig::header::Preset,
+    C: Codec = varsig::encoding::Preset,
+> {
+    pub subject: Subject<DID>,
+    pub issuer: DID,
+    pub audience: DID,
+    pub command: String,
+
+    pub codec: PhantomData<C>,
```

Comment:
Do we absolutely need the `codec` `PhantomData`? Can't rust figure out that it's "inside" the `V: varsig::Header<C>`? Just double-checking.

Reply:
Yeah, I'm also not happy about it. I've tried a bunch of things, but I don't think it's possible. Open to ideas!

Reply:
Yeah. Weird. It's not needed in `Delegation::try_sign`. I guess the inference in functions is better than in structs. :/